templateflow/conf/__init__.py

```diff
@@ -42,14 +42,14 @@
         _update_s3(TF_HOME, local=True, overwrite=True)
 
 
-def update(local=False, overwrite=True):
+def update(local=False, overwrite=True, silent=False):
     """Update an existing DataLad or S3 home."""
     if TF_USE_DATALAD and _update_datalad():
```

Comment:
have you checked this update thing with the _datalad operation mode_? - possibly for a further PR of refinements of that use-case

Reply:
nope - though just did and:
```
Updating TEMPLATEFLOW_HOME using DataLad ...
[ERROR  ] path not associated with any dataset [update(/Users/mathiasg/.cache/templateflow)] 
/Users/mathiasg/code/templateflow/templateflow/conf/__init__.py:74: UserWarning: Error updating TemplateFlow's home directory (using DataLad): Command did not complete successfully [{'action': 'update', 'path': '/Users/mathiasg/.cache/templateflow', 'type': 'directory', 'raw_input': True, 'orig_request': '/Users/mathiasg/.cache/templateflow', 'status': 'error', 'message': 'path not associated with any dataset'}]
  warn(f"Error updating TemplateFlow's home directory (using DataLad): {e}")
```

so I'll commit your changes to avoid the auto-update if the user prefers DataLad.